crates/dbsp/src/algebra/mod.rs

```diff
@@ -342,6 +345,15 @@ impl MulByRef<isize> for Decimal {
     }
 }
 
+impl MulByRef<isize> for SQLDecimal {
+    type Output = Self;
+
+    #[inline]
+    fn mul_by_ref(&self, w: &isize) -> Self::Output {
+        *self * SQLDecimal::from(*w)
```

Comment:
What is the precision and scale of the result?
I think you will need to get them from self first, then use them to convert w, and then do the multiplication